### PR TITLE
Update default value for tech fuel

### PR DESCRIPTION
**Pull request by sahma19**
First we check, if it is able to generate new certificates with the new default value

### CODE REVIEW

File: domains/certificates/Query.API/API.IntegrationTests/ContractTests.cs [hunk 1]
```diff
@@ -166,6 +167,55 @@ public async Task CreateContract_MeteringPointNotOwnedByUser_BadRequest()
...
+        using var response = await client.PostAsJsonAsync("api/certificates/contracts", body);
+
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+
+        var createdContractUri = response.Headers.Location;
+        var createdContract = await client.GetFromJsonAsync<Contract>(createdContractUri);
+
+        createdContract!.Technology.Should().NotBeNull();
+        createdContract.Technology!.FuelCode.Should().Be(technology.AibFuelCode);
+        createdContract.Technology.TechCode.Should().Be(technology.AibTechCode);
```
Comment (duizer) [inline domains/certificates/Query.API/API.IntegrationTests/ContractTests.cs:216]: Can also be a one-liner, something like

```suggestion
        createdContract!.Technology.Should().Be(technology);
```

File: domains/certificates/Query.API/API.IntegrationTests/Shared/Migrations/UpdateTechnologyCodesMigrationTests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        await InsertContractWithEmptyTechnologyFields(dbContext, contractId, "123456789", "DK1");
+
+        var contractBeforeMigration = await dbContext.Contracts.FindAsync(contractId);
+        contractBeforeMigration!.Technology.FuelCode.Should().Be("");
+        contractBeforeMigration.Technology.TechCode.Should().Be("");
+
+        var applyMigration = () => migrator.MigrateAsync("20231114121811_UpdateEmptyTechnologyCodes");
+        await applyMigration.Should().NotThrowAsync();
+
+        await using var newDbContext = await CreateNewCleanDatabase();
```
Comment (duizer) [inline domains/certificates/Query.API/API.IntegrationTests/Shared/Migrations/UpdateTechnologyCodesMigrationTests.cs:37]: This method does not create a new clean database, so this is really hard to read/understand. Also, I am not sure why you want a new dbContext here...

File: domains/certificates/Query.API/API/Migrations/20231114121811_UpdateEmptyTechnologyCodes.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    SET
+        ""Technology_FuelCode"" = CASE
+            WHEN ""Technology_FuelCode"" = '' THEN 'F00000000'
+            ELSE ""Technology_FuelCode""
+        END,
+        ""Technology_TechCode"" = CASE
+            WHEN ""Technology_TechCode"" = '' THEN 'T070000'
+            ELSE ""Technology_TechCode""
+        END
+    WHERE ""Technology_FuelCode"" = '' OR ""Technology_TechCode"" = '';";
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/certificates/Query.API/API/Migrations/20231114121811_UpdateEmptyTechnologyCodes.cs:20]: That is correct. Good Catch.

File: domains/certificates/Query.API/API.IntegrationTests/Shared/Migrations/UpdateTechnologyCodesMigrationTests.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+using Xunit;
+
+namespace API.IntegrationTests.Shared.Migrations;
+
+public class UpdateNullTechnologyCodesMigrationTests : IAsyncDisposable
+{
+    private readonly PostgresContainer container = new();
+
+    [Fact]
+    public async Task ApplyMigration_WhenContractsHaveEmptyTechnologyCodes_UpdatesFieldsCorrectly()
```
Comment (duizer) [inline domains/certificates/Query.API/API.IntegrationTests/Shared/Migrations/UpdateTechnologyCodesMigrationTests.cs:20]: We should have tests for both production and consumption metering point types in the contracts

**Review by duizer — APPROVED** [review]

**Pull request merged**